src/microbe_stage/editor/upgrades/ChemoreceptorUpgradeGUI.cs

```diff
@@ -89,4 +89,25 @@ public void ApplyChanges(MicrobeEditor editor)
         storedOrganelle.SetCustomUpgradeObject(new ChemoreceptorUpgrades(shownChoices[compounds.Selected],
             (float)maximumDistance.Value, (float)minimumAmount.Value, colour.Color));
     }
+
+    public void CompoundChanged(int index)
+    {
+        if (shownChoices?[index] != null && shownChoices != null)
+        {
+            // If the color is in the shownChoices list don't change the color
+            bool isColorInCompundList = false;
+            foreach (Compound compound in shownChoices)
```

Comment:
I think this could be pretty easily be converted to a single LINQ statement which are preferred over these short explicit loops, something like (I wrote this on github so it might not compilte): `bool isCompoundsColourSelected = shownChoices.Any(c => c.Colour == colour.Color)`